src/lava/proc/io/bridge/extractor.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+import typing as ty
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import InPort
+
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.model.py.ports import PyOutPort, PyInPort
+
+from lava.magma.compiler.channels.pypychannel import PyPyChannel
+from lava.magma.runtime.message_infrastructure.multiprocessing import \
+    MultiProcessing
+
+from lava.proc.io.bridge.utils import ChannelConfig, ChannelSendBufferFull, \
+    ChannelRecvBufferEmpty, ChannelRecvBufferNotEmpty,\
+    validate_shape, validate_buffer_size, validate_channel_config, \
+    send_data_blocking, send_data_non_blocking_drop, \
+    recv_empty_blocking, recv_empty_non_blocking_zeros, recv_not_empty_fifo, \
+    recv_not_empty_accumulate
+
+
+class Extractor(AbstractProcess):
+    """Process that gets data out of Lava.
+
+    The Extractor Process exposes a method, recv_data, which enables users to
+    receive data in external Python scripts from the ProcessModel of the
+    Process, while the Lava Runtime is running.
+    Internally, this Process builds a channel (extractor_channel, of type
+    PyPyChannel).
+    The src_port of the channel lives in the ProcessModel.
+    The dst_port of the channel lives in the Process.
+
+    In the ProcessModel, data is received from this Process's InPort,
```

Comment:
The rest of this docstring is probably longer than needed. Just summarize what it does, not exactly how it does it.

Reply:
Made the docstrings shorter.

However, I left a small part explaining how it is doing what it is doing for clarity. I could remove it as well if you still think it is not necessary.